portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -160,4 +160,21 @@ private String getUploadedFileUrl(HttpServletRequest request, String formInputEl
       return imagesService.getServingUrl(options);
     }
   }
+
+  /**
+   * Returns the nickname of the user with id, or empty String if the user has not set a nickname.
+   */
+  private String getUserNickname(String id) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query =
+        new Query("UserInfo")
+            .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    if (entity == null) {
+      return "";
+    }
+    String nickname = (String) entity.getProperty("nickname");
+    return nickname;
```

Comment:
You could consider condensing these two lines so that you are not allocating an extra unused variable:

```suggestion
    return (String) entity.getProperty("nickname");
```

Reply:
Done!